src/components/auth/RegisterParent.vue

```diff
@@ -0,0 +1,242 @@
+          </small>
+        </div>
+        <!--Confirm Password-->
+        <div>
+          <div>
+            <label for="confirmPassword">Confirm Password <span class="required p-1">*</span></label>
+            <PvPassword
+:id="`confirmPassword-${isRegistering ? 'register' : 'login'}`" v-model="v$.confirmPassword.$model" name="confirmPassword"
+              :class="{ 'p-invalid': v$.confirmPassword.$invalid && submitted }" toggle-mask show-icon="pi pi-eye-slash" hide-icon="pi pi-eye" :feedback="false">
+            </PvPassword>
+          </div>
+          <small v-if="(v$.confirmPassword.$invalid && submitted) || v$.confirmPassword.$pending.$response" class="p-error">
+            Passwords must match
+          </small>
+        </div>
+      </section>
+      <!--Accept Checkbox-->
+      <section class="form-section flex lg:flex-row">
+        <!-- Recaptcha + consent -->
+        <ChallengeV3 v-model="response" action="submit">
+          <div class="field-checkbox terms-checkbox">
+            <PvCheckbox
+:id="`accept-${isRegistering ? 'register' : 'login'}`" v-model="v$.accept.$model" name="accept" binary
+              :disabled="showConsent" :class="{ 'p-invalid': v$.accept.$invalid && submitted }" @change="getConsent"/>
+            <label for="accept" :class="{ 'p-error': v$.accept.$invalid && submitted }">I agree to the terms and conditions<span class="required">*</span></label>
+          </div>
+          <small v-if="(v$.accept.$invalid && submitted) || v$.accept.$pending.$response" class="p-error">
+              You must agree to the terms and conditions
+          </small>
+        </ChallengeV3>
+      </section>
+      <ConsentModal v-if="showConsent" :consent-text="consentText" consent-type="consent" @accepted="handleConsentAccept" />
+      <div class="form-submit">
+        <PvButton type="submit" label="Next" class="submit-button" :disabled="isNextButtonDisabled"/>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script setup>
+import { computed, reactive, ref, defineEmits } from "vue";
+import { required, sameAs, minLength, } from "@vuelidate/validators";
+import { useVuelidate } from "@vuelidate/core";
+import { useAuthStore } from "@/store/auth";
+import ConsentModal from "../ConsentModal.vue";
+import { ChallengeV3 } from 'vue-recaptcha';
+
+
+const authStore = useAuthStore()
+const isCaptchaverified = ref(null);
+
+const props = defineProps({
+  isRegistering: {type: Boolean, default: true},
+});
+
+console.log(props);
+
+const emit = defineEmits(['submit']);
+
+const state = reactive({
+  // activationCode: "",
+  firstName: "",
+  lastName: "",
+  password: "",
+  confirmPassword: "",
+  accept: false,
+});
+const passwordRef = computed(() => state.password);
+
+
+
+const ParentEmail = (value) => {
+  if (!value.includes('@')) return true;
+
+  const emailRegex = /^(?!.*@.*@)[a-zA-Z0-9]+([._]?[a-zA-Z0-9]+)*(@[a-zA-Z0-9]+([.-]?[a-zA-Z0-9]+)*(\.[a-zA-Z]{2,7})+)?$/;
```

Comment:
I think we don't even need this, we just have to add on the input the type email and the form will do all the validation for us since we are using html5.